src/video/cameradevice.cpp

```diff
@@ -55,7 +55,10 @@ CameraDevice* CameraDevice::open(QString devName, AVDictionary** options)
     CameraDevice* dev = openDevices.value(devName);
     int aduration;
     if (dev)
-        goto out;
```

Comment:
No, in this case the goto ensures that in every case where the function is left the lock is unlocked, and removing all the gotos also causes a lot of code duplication.

Both introduce unnecessary possibilites for new mistakes, which could be avoided by that goto.